src/Reflection/Adapter/ReflectionParameter.php

```diff
@@ -185,19 +174,16 @@ public function getDefaultValueConstantName()
         return $this->betterReflectionParameter->getDefaultValueConstantName();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    public function hasType()
+    public function hasType(): bool
     {
         return $this->betterReflectionParameter->hasType();
     }
 
     /**
-     * {@inheritDoc}
+     * @psalm-suppress MismatchingDocblockReturnType
      */
-    public function getType()
+    public function getType(): ReflectionNamedType|ReflectionUnionType|ReflectionType|null
```

Comment:
IMO we should not return `ReflectionType`, but rather just one of `ReflectionNamedType|ReflectionUnionType|null`.

At a later point in time, this highlights the BC break of PHP 8.1, which expanded this union type again ( @CiaranMcNulty would have appreciated this at static analysis level )